Web.md

```diff
@@ -202,6 +202,92 @@ Given these values, to process a potential WebAssembly response:
+  2. If *targetRealm*'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     *serialized*.[[AgentCluster]], then throw a
+     "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+     [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+  3. Set *value* to a new WebAssembly.Memory object in *targetRealm*, where:
+     1. *value*'s internal slot [[WebAssembly.Memory]] is set to a new `Memory.memory` *mem*, where:
+        1. *mem*'s linear memory is set to *serialized*.[[BufferObject]].[[ArrayBufferData]].
+        2. *mem*'s linear memory has a byte length of *serialized*.[[BufferObject]].[[ArrayBufferByteLength]].
+        3. *mem*'s `max` size is set tof *serialized*.[[MemoryMaximum]].
+     2. *value*'s internal slot [[BufferObject]] is set to a new `SharedArrayBuffer` *buf*, where:
+        1. *buf*'s internal slot [[ArrayBufferData]]
+           aliases *mem*'s linear memory.
+        2. *buf*'s internal slot [[ArrayBufferByteLength]]
+           is set to the byte length of *mem*'s linear memory.
+
+
+### Structured serialization of a `WebAssembly.Module`
+
+A `WebAssembly.Module` is a
+[serializable object](https://html.spec.whatwg.org/multipage/infrastructure.html#serializable-objects), and will require changes to the
+[HTML Standard](https://html.spec.whatwg.org/multipage/infrastructure.html#serialization-steps) as follows:
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if *value* has a [[WebAssembly.Module]] internal slot, then:
+  1. If *forStorage* is true and the relevant settings
+     object of *value* is not "Secure" per the
+     ["Is the environment settings object a secure context?"](https://w3c.github.io/webappsec-secure-contexts/#settings-object)
+     algorithm, throw a
+     "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+     [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
+  2. Otherwise,
+     set *serialized* to { [[Type]]: "WebAssembly.Module",
+     [[WebAssembly.Module]]: *value*.[[WebAssembly.Module]],
+     [[AgentCluster]]: the [current Realm Record](https://tc39.github.io/ecma262/#current-realm)'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) }.
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if *serialized*.[[Type]] is "WebAssembly.Module", then:
+  1. If *targetRealm*'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     *serialized*.[[AgentCluster]], then throw a
+     "[`DataCloneError`](https://heycam.github.io/webidl/#datacloneerror)"
+     [`DOMException`](https://heycam.github.io/webidl/#dfn-DOMException).
```

Comment:
Gotcha.  So my root question was whether this definition was permissive enough to allow cross-origin code caching (e.g., in the local IDB of a CDN origin), bit it looks like 'no', that'll have to wait for foreign-fetch...